pandas/tests/arrays/floating/test_arithmetic.py

```diff
@@ -0,0 +1,234 @@
+        s = pd.Series(data)
+        ops = getattr(s, op)
+        opa = getattr(data, op)
+
+        # invalid scalars
+        msg = (
+            r"(:?can only perform ops with numeric values)"
+            r"|(:?FloatingArray cannot perform the operation mod)"
+        )
+        with pytest.raises(TypeError, match=msg):
+            ops("foo")
+        with pytest.raises(TypeError, match=msg):
+            ops(pd.Timestamp("20180101"))
+
+        # invalid array-likes
+        with pytest.raises(TypeError, match=msg):
+            ops(pd.Series("foo", index=s.index))
+
+        if op != "__rpow__":
+            # TODO(extension)
+            # rpow with a datetimelike coerces the integer array incorrectly
+            msg = (
+                "can only perform ops with numeric values|"
+                "cannot perform .* with this index type: DatetimeArray|"
+                "Addition/subtraction of integers and integer-arrays "
+                "with DatetimeArray is no longer supported. *"
+            )
+            with pytest.raises(TypeError, match=msg):
+                ops(pd.Series(pd.date_range("20180101", periods=len(s))))
+
+        # 2d
+        result = opa(pd.DataFrame({"A": s}))
+        assert result is NotImplemented
+
+        msg = r"can only perform ops with 1-d structures"
+        with pytest.raises(NotImplementedError, match=msg):
+            opa(np.arange(len(s)).reshape(-1, len(s)))
+
+    @pytest.mark.parametrize("zero, negative", [(0, False), (0.0, False), (-0.0, True)])
+    def test_divide_by_zero(self, zero, negative):
+        # TODO pending NA/NaN discussion
+        # https://github.com/pandas-dev/pandas/issues/32265/
+        a = pd.array([0, 1, -1, None], dtype="Float64")
+        result = a / zero
+        expected = FloatingArray(
+            np.array([np.nan, np.inf, -np.inf, np.nan]),
+            np.array([False, False, False, True]),
+        )
+        if negative:
+            expected *= -1
+        tm.assert_extension_array_equal(result, expected)
+
+    def test_pow_scalar(self):
+        a = pd.array([-1, 0, 1, None, 2], dtype="Float64")
+        result = a ** 0
+        expected = pd.array([1, 1, 1, 1, 1], dtype="Float64")
+        tm.assert_extension_array_equal(result, expected)
+
+        result = a ** 1
+        expected = pd.array([-1, 0, 1, None, 2], dtype="Float64")
+        tm.assert_extension_array_equal(result, expected)
+
+        result = a ** pd.NA
+        expected = pd.array([None, None, 1, None, None], dtype="Float64")
+        tm.assert_extension_array_equal(result, expected)
+
+        result = a ** np.nan
+        # TODO np.nan should be converted to pd.NA / missing before operation?
+        expected = FloatingArray(
+            np.array([np.nan, np.nan, 1, np.nan, np.nan], dtype="float64"), mask=a._mask
+        )
+        tm.assert_extension_array_equal(result, expected)
+
+        # reversed
+        a = a[1:]  # Can't raise integers to negative powers.
```

Comment:
Shouldn't this be possible, with the result now being a FloatingArray? Do we also want to test raising float values (other than NaN) to a FloatingArray power?